src/Data/Hashable/Class.hs

```diff
@@ -840,18 +838,22 @@ hashByteArrayWithSalt
     -> Int         -- ^ salt
     -> Int         -- ^ hash value
 hashByteArrayWithSalt ba !off !len !h =
-    fromIntegral $ c_hashByteArray ba (fromIntegral off) (fromIntegral len)
-    (fromIntegral h)
+    fromIntegral $
+    c_siphash24_offset k0 (fromSalt h) ba (fromIntegral off) (fromIntegral len)
 
-#if __GLASGOW_HASKELL__ >= 802
-foreign import capi unsafe "HsHashable.h hashable_fnv_hash_offset" c_hashByteArray
-#else
-foreign import ccall unsafe "hashable_fnv_hash_offset" c_hashByteArray
-#endif
+k0 :: Word64
+k0 = 0x56e2b8a0aee1721a
```

Comment:
what are these magic constants?

Reply:
these are the `k` that is supposed to be hidden from an adversary. I left them in for testing, but now they're flagged with a big TODO comment as well.

Reply:
If we can hide the initial seed then FNV is a good hash too, isn't it?

Reply:
For FNV it's possible to create universal collision payloads, payloads that collide regardless of salt: https://medium.com/@robertgrosse/generating-64-bit-hash-collisions-to-dos-python-5b21404a5306

In that case he's cheating a little by creating sets of 64 keys, but it works, and a determined attacker could probably make a more efficient implementation.

It's better to hide the salt in FNV, it makes attacking slightly more difficult, but it doesn't solve the problem.
I wouldn't have attempted re-adding siphash if I could hide the salt for FNV. (I tried to)

Reply:
I always forget about universal collisions, thanks for the reminder